doc/embedded/crosscompile_armv7.md

```diff
@@ -0,0 +1,188 @@
+## Run mlpack bindings on Raspberry PI
+
+In this article, we explore how to run mlpack command-line programs on a
+Raspberry Pi 2.  mlpack is a great choice for machine learning tasks on embedded
+and low-resource hardware (like the Raspberry Pi) due to its design as a
+lightweight header-only C++ library with a focus on efficiency.
+
+<!-- TODO: link to kNN documentation when it's done -->
+
+mlpack provides convenient command-line bindings for many of its algorithms.
+In this tutorial, we are going to compile the k-nearest-neighbors command-line
+program (kNN), and then run it on a Pi 2.
+
+Since the Pi 2 has 1 GB of RAM, compilation of mlpack code directly on the Pi
+can be challenging.  Depending on the OS you are using, 250 MB or more of the
+RAM may already be in use just by the system.  This leaves us with around 750 MB
+of RAM, and given that the Pi 2 has only ARMv7 core clocked at 1 GHz,
+it might take a lot of time for the compilation to finish. As a result it will
+be much easier to cross-compile mlpack on a more powerful development system,
+and then move the compiled code directly to the Raspberry Pi.
+
+The first step is to download a cross compiler toolchain that is going to run on
+an `x86_64` machine (the host) and produce binaries that runs on an ARMv7
+machine (this is the Raspberry Pi, the target).  There are several toolchains
+around, but the [Bootlin](https://bootlin.com/) toolchains are probably the
+easiest to set up and use.  Bootlin has produced toolchains for all kind of
+architectures (ARMv7, ARMv8, Sparc, MIPS, RISC-V, etc.), with three different
+sets of compilers. To get the toolchain please visit the following website:
+[`https://toolchains.bootlin.com/`](https://toolchains.bootlin.com/).
+
+You need to select the architecture and the compiler set. In our case, we are
+looking for the arch `armv7-eabihf`, and the libc `glibc`.  You can also
+use `musl` or `uclibc` if you are interested in reducing the size of the
+compiled programs.  However, in this tutorial, we will stick to glibc for
+simplicity.
+
+If you want to get a link directly to the toolchain and download it with `wget`
+use the following command:
+
+```sh
+wget https://toolchains.bootlin.com/downloads/releases/toolchains/armv7-eabihf/tarballs/armv7-eabihf--glibc--stable-2024.02-1.tar.bz2
+```
+
+Once we have downloaded the toolchain we can extract the content using the
+following command:
+
+```sh
+tar -xvf armv7-eabihf--glibc--stable-2024.02-1.tar.bz2
+```
+
+Once we have done that, now the toolchain is ready.  Feel free to explore inside
+the toolchain; you will find the basic set of compilers in addition to a
+`sysroot/` folder that contains headers for standard libraries.
+
+Now, we will use this toolchain to compile mlpack. The first thing is to get the
+mlpack sources.  You can either download the archive file of the latest version,
```

Comment:
I can make a link `https://www.mlpack.org/files/mlpack-latest.tar.gz` that we could link to here---what do you think?  We have that for ensmallen.